src/basic/env-util.c

```diff
@@ -336,7 +349,7 @@ char **strv_env_unset(char **l, const char *p) {
 
         for (f = t = l; *f; f++) {
 
-                if (env_match(*f, p)) {
+                if (env_entry_has_name(*f, p)) {
```

Comment:
OK, thanks for the explanation. I was wondering why this is like this.

All APIs to read the environment entries (except for manually iterating over the third argument to `main`, but that's special) assume that keys are unique. So we should always keep the keys unique.

I'll change the `set` and `replace` functions to do that.